### PR TITLE
Improve script functions test coverage

### PR DESCRIPTION
**Pull request by zas**
# Summary

* This is a…
    * [x] Bug fix
    * [ ] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [x] Other
* **Describe this change in 1-2 sentences**:

# Problem

Coverage of script functions code is incomplete

* JIRA ticket (_optional_): PICARD-XXX

# Solution

Improve tests and code so everything is covered, including corner-cases.
After these changes coverage is 99%, and 100% on script functions code.

# Action

### CODE REVIEW

File: test/test_script.py [hunk 1]
```diff
@@ -990,9 +1045,9 @@ def test_cmd_datetime(self):
             self.assertScriptResultEquals("$datetime(%foo%)", "20200102123456.000789", context)
             self.assertScriptResultEquals("$datetime($initials())", "2020-01-02 12:34:56", context)
             # Tests with static input
-            self.assertScriptResultEquals("$datetime(\%Y\%m\%d\%H\%M\%S.\%f)", "20200102123456.000789", context)
+            self.assertScriptResultEquals("$datetime(\\%Y\\%m\\%d\\%H\\%M\\%S.\\%f)", "20200102123456.000789", context)
             # Tests with timezones
-            self.assertScriptResultEquals("$datetime(\%H\%M\%S \%z \%Z)", "123456 +0200 TZ Test", context)
+            self.assertScriptResultEquals("$datetime(\\%H\\%M\\%S \\%z \\%Z)", "123456 +0200 TZ Test", context)
```
Comment (phw) [inline test/test_script.py]: Maybe better use `r"$datetime(\%H\%M\%S \%z \%Z)"` instead, same above.

Comment (zas) [reply test/test_script.py]: Done in ca148ee

**Comment by phw** [comment]
@zas I tested the `$div(n,0)` case before this patch and it indeed you can construct scripts that just make Picard crash. I added PICARD-1757 for this to have it in the changelog. Great catch, and shows again how valuable tests are.
